include/regs.h

```diff
@@ -166,4 +166,141 @@
+#define R_PLAY_RUN_UPDATE             HREG(81)
+#define R_PLAY_RUN_DRAW               HREG(82)
+#define R_PLAY_DRAW_SKYBOX            HREG(83)
+#define R_PLAY_DRAW_ROOM_FLAGS        HREG(84) // also enables drawing the scene config and room bg cover
+#define R_PLAY_DRAW_ACTORS            HREG(85)
+#define R_PLAY_DRAW_LENS_FLARES       HREG(86)
+#define R_PLAY_DRAW_SCREEN_FILLS      HREG(87)
+#define R_PLAY_DRAW_SANDSTORM         HREG(88)
+#define R_PLAY_DRAW_OVERLAY_ELEMENTS  HREG(89)
+#define R_PLAY_DRAW_ASSORTED_FLAGS    HREG(90) // see flags below
+#define R_PLAY_DRAW_SKYBOX_FILTERS    (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 0))
+#define R_PLAY_DRAW_SUN_AND_MOON      (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 1))
+#define R_PLAY_DRAW_LIGHTNING         (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 2))
+#define R_PLAY_DRAW_LIGHTS            (R_PLAY_DRAW_ASSORTED_FLAGS & (1 << 3))
+#define R_PLAY_DRAW_COVER_ELEMENTS    HREG(92) // draw transition instance and vismono
+#define R_PLAY_DRAW_DEBUG_OBJECTS     HREG(93)
+
+#define R_PLAY_ENABLE_UCODE_DISAS    HREG(93) // set to 2 to switch to ucode disas mode and trigger it
+#define R_PLAY_UCODE_DISAS_LOG_LEVEL HREG(92) // log level to use when switching to ucode disas
+
+// HREG_MODE_PERSPECTIVE
+#define R_PERSPECTIVE_INIT    HREG(94) // set to `HREG_MODE_PERSPECTIVE` when init is complete
+#define R_PERSPECTIVE_FOVY    HREG(83) // value to use for `fovy`
+#define R_PERSPECTIVE_ASPECT  HREG(84) // value to use for `aspect`, divided by 10000
+#define R_PERSPECTIVE_NEAR    HREG(85) // value to use for `near`
+#define R_PERSPECTIVE_FAR     HREG(86) // value to use for `far`
+#define R_PERSPECTIVE_SCALE   HREG(87) // value to use for `scale`, divided by 100
+
+// HREG_MODE_INPUT_TEST
+#define R_INPUT_TEST_CONTROLLER_PORT      HREG(81) // controller to copy values from
+#define R_INPUT_TEST_COMPARE_VALUE        HREG(82) // input value compared against for various tests
+#define R_INPUT_TEST_BUTTON_CUR           HREG(83) // buttons held on the test controller
+#define R_INPUT_TEST_BUTTON_PRESS         HREG(84) // buttons pressed on the test controller
+#define R_INPUT_TEST_REL_STICK_X          HREG(85) // current stick value on the x axis (ignoring the deadzone)
+#define R_INPUT_TEST_REL_STICK_Y          HREG(86) // current stick value on the y axis (ignoring the deadzone)
+#define R_INPUT_TEST_REL_STICK_X_2        HREG(87) // same as R_INPUT_TEST_REL_STICK_X
+#define R_INPUT_TEST_REL_STICK_Y_2        HREG(88) // same as R_INPUT_TEST_REL_STICK_Y
+#define R_INPUT_TEST_CUR_STICK_X          HREG(89) // current stick value on the x axis
+#define R_INPUT_TEST_CUR_STICK_Y          HREG(90) // current stick value on the y axis
+#define R_INPUT_TEST_COMPARE_BUTTON_CUR   HREG(93) // set to 1 if cur.button matches R_INPUT_TEST_COMPARE_VALUE
+#define R_INPUT_TEST_COMPARE_COMBO_CUR    HREG(94) // set to 1 if cur.button combo matches R_INPUT_TEST_COMPARE_VALUE
```

Comment:
I dont fully follow, doesnt `R_INPUT_TEST_COMPARE_COMBO_CUR` need all of the buttons to match because it uses `CHECK_BTN_ALL`? An "at least" situation would need `CHECK_BTN_ANY` right?

Reply:
i had to print the values out and watch it to understand, but yeah i see it now. see if my new comments help

Reply:
looks fine :)